src/main/java/org/simplify4u/plugins/skipfilters/ReactorDependencySkipper.java

```diff
@@ -46,16 +56,58 @@ public boolean shouldSkipArtifact(Artifact artifact) {
      *          The to check against upstream reactor dependencies.
      *
      * @return  {@code true} if the specified artifact is in the current Maven reactor build and is
-     *          a direct or transitive dependency of the current project; {@code false} if it is not
-     *          either.
+     *          a direct or transitive dependency of the current project; {@code false} if it is
+     *          not either.
      */
     private boolean isUpstreamReactorDependency(final Artifact artifact) {
         for (final MavenProject upstreamProject : this.upstreamProjects) {
-            if (upstreamProject.getId().equals(artifact.getId())) {
+            if (this.artifactMatchesProject(artifact, upstreamProject)) {
                 return true;
             }
         }
 
         return false;
     }
+
+    /**
+     * Check whether the specified artifact belongs to the specified Maven project.
+     *
+     * @param artifact
+     *   The artifact being checked.
+     * @param project
+     *   The project to which the artifact will be compared.
+     *
+     * @return
+     *   {@code true} if the artifact was produced by the specified project; or
+     *   {@code false} if it is not.
+     */
+    private boolean artifactMatchesProject(final Artifact artifact,
+                                           final MavenProject project) {
+        final Artifact projectArtifact = project.getArtifact();
+
+        return projectArtifact != null
+               && this.artifactsMatch(projectArtifact, artifact);
+    }
+
+    /**
+     * Check whether the specified artifacts are equivalent, or at least
+     * identify the same group, artifact ID, and version.
+     *
+     * @param artifact1
+     *   The first artifact.
+     * @param artifact2
+     *   The second artifact.
+     *
+     * @return
+     *   If either the two artifacts are equal or correspond to the same Maven
+     *   coordinates.
+     */
+    @SuppressWarnings("PMD.UselessParentheses")
```

Comment:
is it need?

Reply:
yes. PMD flagged the fact that I'm grouping the AND conditions with parens separately from the single OR on line 109 as an "unnecessary use of parens". I get the fact that the precedence order of `&&` is higher than `||`, but I wanted to be explicit that we're going to try the `.equals()` check first, and then if that does not pass, move on to comparing group, artifact ID, and version as the next logical group.